stdsimd/arch/detect/linux/auxvec.rs

```diff
@@ -0,0 +1,290 @@
+//! Parses ELF auxiliary vectors.
+
+use core::mem;
+use _std::prelude::v1::*;
+use _std::fs::File;
+use _std::io::Read;
+
+#[cfg(not(all(target_os = "linux",
+              any(target_arch = "aarch64", target_arch = "arm",
+                  target_arch = "powerpc64", target_arch = "mips",
+                  target_arch = "mips64"))))]
+compile_error!("ELF auxiliary vectors are not implemented for this target.");
+
+/// Key to access the CPU Hardware capabilities bitfield.
+#[cfg(any(target_arch = "aarch64", target_arch = "arm",
+          target_arch = "powerpc64", target_arch = "mips",
+          target_arch = "mips64"))]
+pub const AT_HWCAP: usize = 16;
+/// Key to access the CPU Hardware capabilities 2 bitfield.
+#[cfg(any(target_arch = "arm", target_arch = "powerpc64"))]
+pub const AT_HWCAP2: usize = 26;
+
+/// Cache HWCAP bitfields of the ELF Auxiliary Vector.
+///
+/// If an entry cannot be read all the bits in the bitfield are set to zero.
+/// This should be interpreted as all the features being disabled.
+#[derive(Debug, Copy, Clone)]
+pub struct AuxVec {
+    #[cfg(any(target_arch = "aarch64", target_arch = "arm",
+              target_arch = "powerpc64", target_arch = "mips",
+              target_arch = "mips64"))]
+    pub hwcap: usize,
+    #[cfg(any(target_arch = "arm", target_arch = "powerpc64"))]
+    pub hwcap2: usize,
```

Comment:
Could both fields be perhaps unconditionally included to avoid the extra `#[cfg]` traffic?